FarihaIS/VanRoomies
# Fix bidirectional exclusion bug and allow blocking/reporting

# Welcome to VanRoomies! 👋✈️

Fixes: #102 

## Description of the change:
- Bugfix for user exclusion on recommendations after matching. Via this PR when a user matches with another user, now both these users will not show up on each others list of recommendations.  Prior this change was only happening on one side, so that the other user was still getting a set of recommendations which included their match.

- Additionally, similar to the above functionality this PR adds support for user blocking which is done based on a threshold. This requires a minor change to the User schema so that a user document tracks how many times it has been blocked and once its block count exceeds the block threshold, the user is removed from the app alongside any other information relating to them - such as preferences or listings.

- Lastly, we also want to allow users to report listings as potential scams. Similar to the above, if the count of reports for a listings reaches a certain threshold we remove that listing. Also once a user reports a listing as scam, that listing should no longer show up as a recommended one for that user.

## How to manually test:
1. Run `npm run dev`
2. If you do not have users that also have correspoding preferences set make sure to create at least two users and set correspoding preferences for each, call them user A and user B
3. From Postman use the `PUT Update Exclude User` route where the ID of one user goes in the URL and the ID of the other user goes on the request body. Once the request a sent, you should be able to see in MongoDB compass that each user has been added to the list `notRecommended` for the other user.
